iroh-bitswap/src/message.rs

```diff
@@ -16,31 +16,144 @@ mod pb {
 /// Priority of a wanted block.
 pub type Priority = i32;
 
-/// A bitswap message.
-#[derive(Default, Clone, PartialEq)]
-pub struct BitswapMessage {
+#[derive(Default, Clone, Debug, PartialEq)]
+pub struct Wantlist {
     /// Wanted blocks.
-    pub(crate) want: HashMap<Cid, Priority>,
+    want_blocks: AHashMap<Cid, Priority>,
+    /// Wanted haves.
+    want_haves: AHashSet<Cid>,
```

Comment:
tbh this is a pretty confusing (but logical) name. Maybe the comment should read something more like `/// Set of Cids for blocks that are on the want list, but that we already have`?

Reply:
removed it for now as we are not using it anyway and it is a very confusing thing